docs/sources/flow/reference/components/otelcol.exporter.otlp.md

```diff
@@ -0,0 +1,199 @@
+[queue]: #queue-block
+[retry]: #retry-block
+
+### client block
+
+The `client` block configures the gRPC client used by the component.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`endpoint` | `string` | `host:port` to send telemetry data to. | | yes
+`compression` | `string` | Compression mechanism to use for requests. | `"gzip"` | no
+`read_buffer_size` | `string` | Size of the read buffer the gRPC client to use for reading server responses. | | no
+`write_buffer_size` | `string` | Size of the write buffer the gRPC client to use for writing requests. | `"512KiB"` | no
+`wait_for_ready` | `boolean` | Waits for gRPC connection to be in the `READY` state before sending data. | `false` | no
+`headers` | `map(string)` | Additional headers to send with the request. | `{}` | no
+`balancer_name` | `string` | Which gRPC client-side load balancer to use for requests. | | no
+
+By default, requests are compressed with gzip. The `compression` argument
+controls which compression mechanism to use. Supported strings are:
+
+* `"gzip"`
+* `"zlib"`
+* `"deflate"`
+* `"snappy"`
+* `"zstd"`
+
+If `compression` is set to `"none"` or an empty string `""`, no compression is
+used.
+
+The `balancer_name` argument controls what client-side load balancing mechanism
+to use. See the gRPC documentation on [Load balancing][] for more information.
+When unspecified, `pick_first` is used.
+
+[Load balancing]: https://github.com/grpc/grpc-go/blob/master/examples/features/load_balancing/README.md#pick_first
+
+### tls block
+
+The `tls` block configures TLS settings used for the connection to the gRPC
+server.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`ca_file` | `string` | Path to the CA file. | | no
+`cert_file` | `string` | Path to the TLS certificate. | | no
+`key_file` | `string` | Path to the TLS certificate key. | | no
+`min_version` | `string` | Minimum acceptable TLS version for connections. | `"TLS 1.2"` | no
+`max_version` | `string` | Maximum acceptable TLS version for connections. | `"TLS 1.3"` | no
+`insecure` | `boolean` | Disables TLS when connecting to the gRPC server. | | no
+`insecure_skip_verify` | `boolean` | Ignores insecure server TLS sertificates. | | no
+`server_name` | `string` | Verifies the hostname of server certificates when set. | | no
+
+The `tls` block should always be provided, even if the server doesn't support
+TLS. To disable `tls` for connections to the gRPC server, set the `insecure`
+argument to `true`.
+
+### keepalive block
+
+The `keepalive` block configures keepalive settings for gRPC client
+connections.
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`time` | `duration` | How often to ping the server after no activity. | | no
```

Comment:
`time` feels very generic; how about ping or keepalive? This is probably what its called currently but I feel we can make this clearer. Then we can rename `permit_without_stream` to `allow_ping_without_stream`

Reply:
@mattdurham WDYT about:

* `ping_wait` 
* `pong_timeout` 
* `ping_without_stream`

Reply:
LGTM